src/net/poll.rs

```diff
@@ -70,13 +77,43 @@ pub struct NetworkState {
 }
 
 impl NetworkState {
-    pub fn bind(addr: &SocketAddr, capacity: usize) -> Result<NetworkState, net_error> {
-        let server = mio_net::TcpListener::bind(addr)
-            .map_err(|e| {
-                error!("Failed to bind to {:?}: {:?}", addr, e);
-                net_error::BindError
-            })?;
+    fn bind_address(addr: &SocketAddr) -> Result<mio_net::TcpListener, net_error> {
+        if !cfg!(test) {
+            mio_net::TcpListener::bind(addr)
+                .map_err(|e| {
+                    error!("Failed to bind to {:?}: {:?}", addr, e);
+                    net_error::BindError
+                })
+        }
+        else {
+            let mut backoff = 1000;
+            let mut rng = rand::thread_rng();
+            let mut count = 1000;
+            loop {
+                match mio_net::TcpListener::bind(addr) {
+                    Ok(server) => {
+                        return Ok(server);
+                    },
+                    Err(e) => match e.kind() {
+                        io::ErrorKind::AddrInUse => {
+                            debug!("Waiting {} millis and trying to bind {:?} again", backoff, addr);
+                            sleep_ms(count);
```

Comment:
I think that `sleep_ms(count)` should be using backoff instead of count?

Reply:
Yes, will fix.

Reply:
Should be fixed now.